internal/lm/resource.go

```diff
@@ -90,6 +91,10 @@ func NewMIGResourceLabeler(resourceName spec.ResourceName, config *spec.Config,
 	}
 
 	resourceLabeler := newResourceLabeler(resourceName, config)
+	if strategy := resourceLabeler.sharing.SharingStrategy(); strategy == spec.SharingStrategyMPS {
+		klog.Warningf("Ignoring unsupported sharing strategy for MIG devices: %v", strategy)
```

Comment:
Question: instead of issuing a warning and removing the setting, should we error out GFD instead?

Reply:
The plugin raises an error.

Reply:
What would be the reason to not just do the same here then?

Reply:
This does mean that no other labels are generated. In the case where a user is just deploying GFD -- possibly with a faulty config -- we would end up with no labels at all.

Reply:
You mean running GFD alongside a different device plugin that doesn't support our config file? I still think we would want to "fail-fast" in this case -- prompting an admin to check the logs and update the config file. Unless I am missing some case where updating the config wouldn't be possible, but we want to allow them to continue anyway.